### DISCUSSION TITLE
animated set_viewport_scale issues

### DISCUSSION THREAD

**Discussion by Xavman42** [DigiScore/neoscore#88] [Q&A] [answered]
I'm getting some unexpected behavior when I run the following bit of code - the visuals aren't important, I'm just replicating the bad behavior as simply as I can:
```
from math import sin, pi
from neoscore.common import *


def refresh_func(t: float):
    zoom = 4 + 5 * (sin(t * pi))
    neoscore.set_viewport_scale(zoom)
    neoscore.set_viewport_rotation(10 * t)
    print("zoom calc", zoom)
    print("neo scale", neoscore.get_viewport_scale())


neoscore.setup()

s = Staff(Point(Mm(0), -Mm(0)), None, Mm(100))

neoscore.show(refresh_func)
```

I would expect the two print statements to give me the same number, but `neoscore.get_viewport_scale()` often gives very unexpected results. For example: 
```
zoom calc 7.620290657894332
neo scale 12.636253170256024
zoom calc 7.238295613532763
neo scale 11.911141489407502
zoom calc 6.818400067529553
neo scale 11.135344919438163
```

Sometimes the results are so unexpected that I get a divide-by-zero crash that looks something like this:
```
Traceback (most recent call last):
  File "/my_directory/GuitaRPG/venv/lib/python3.10/site-packages/neoscore/interface/qt/main_window.py", line 45, in refresh
    requested_delay_s = self.refresh_func(start_time)
  File "/my_directory/GuitaRPG/venv/lib/python3.10/site-packages/neoscore/core/neoscore.py", line 485, in wrapped_refresh_func
    refresh_func(frame_time)
  File "/my_directory/GuitaRPG/NeoScore/map.py", line 114, in camera_rotate_refresh_func
    neoscore.set_viewport_scale(4)
  File "/my_directory/GuitaRPG/venv/lib/python3.10/site-packages/neoscore/core/neoscore.py", line 305, in set_viewport_scale
    app_interface.viewport_scale = scale
  File "/my_directory/GuitaRPG/venv/lib/python3.10/site-packages/neoscore/interface/app_interface.py", line 291, in viewport_scale
    relative_scale_factor = value / current_scale
ZeroDivisionError: float division by zero
```

Is this a bug that can be easily fixed, or should I just avoid animating scaling?

**Comment by ajyoon** [comment, marked as answer]
yeah, looks like a bug. I've filed #89 for it. shouldn't be a difficult fix if you want to take a crack at it, otherwise I should be able to work on it in the next week or two.

**Reply by ajyoon** [reply]
actually I took a look and found [the fix was very simple so i just pushed it](7af8feda89d6fc2fda4387e41900ad6f18588970). If you need to use this urgently I can cut a release or you can run off the dev repo if you know how to do that.

**Reply by Xavman42** [reply]
I'm in no rush. Thank you so much!
